inbm/dispatcher-agent/dispatcher/sota/os_factory.py

```diff
@@ -137,10 +139,12 @@ def create_os_updater(self) -> OsUpdater:
         logger.debug("")
         return YoctoX86_64Updater()
 
-    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], proceed_without_rollback: bool) -> Snapshot:
+    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], 
+                           proceed_without_rollback: bool, device_reboot: str) -> Snapshot:
         logger.debug("")
         trtl = Trtl(PseudoShellRunner(), BTRFS)
-        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, proceed_without_rollback)
+        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, 
+                             proceed_without_rollback, device_reboot)
```

Comment:
**Suggestion:** The 'device_reboot' parameter is a string that likely expects 'yes' or 'no' values. It would be safer to validate this input to ensure it meets expectations and handle any invalid values appropriately. [validation]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], 
                           proceed_without_rollback: bool, device_reboot: str) -> Snapshot:
        if device_reboot.lower() not in ['yes', 'no']:
            raise ValueError("Invalid value for device_reboot. Expected 'yes' or 'no'.")
        logger.debug("Creating snapshotter with device_reboot set to {}".format(device_reboot))
        trtl = Trtl(PseudoShellRunner(), BTRFS)
        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, 
                             proceed_without_rollback, device_reboot)
```